lib/core/utils/parse-crossorigin-stylesheets.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Parse cross-origin stylesheets
+ *
+ * @param {String} url url from which to fetch stylesheet
+ * @param {Object} options options object from `axe.utils.parseStylesheets`
+ * @param {Array<Number>} priority sheet priority
+ * @param {Array<String>} importedUrls urls of already imported stylesheets
+ * @param {Boolean} isCrossOrigin boolean denoting if a stylesheet is `cross-origin`
+ * @returns {Promise}
+ */
+axe.utils.parseCrossOriginStylesheets = function parseCrossOriginStylesheets(
+	url,
+	options,
+	priority,
+	importedUrls,
+	isCrossOrigin
+) {
+	const axiosOptions = {
+		method: 'get',
+		url,
+		timeout: options.timeout
+	};
+
+	importedUrls.push(url);
+	return axe.imports.axios(axiosOptions).then(({ data }) => {
+		const result = options.convertDataToStylesheet({
+			data,
+			isCrossOrigin,
+			priority,
+			root: options.rootNode,
+			shadowId: options.shadowId
+		});
+
+		/**
+		 * Note:
+		 * Safety check to stop recursion,
+		 * if there are numerous nested `@import` statements
+		 */
+		if (importedUrls.length > axe.constants.preload.maxImportUrls) {
```

Comment:
Since `importedUrls` is passed around from the start of getting all stylesheets, won't this check only allow you to load 10 files max? So if there are 11 stylesheets, and the first 10 have no import but the 11th one does, we won't load any imports from the 11th file? Also, this prevents a single css file from having more than 10 imports (even if they are all leaf node files) since we won't load the 11th.

Reply:
What do you mean by leaf node files?
Also `importedUrls` is only absolute `urls` and not relative ones.
The number `10` is up for debate. But yes, only first 10 will be loaded.

Reply:
Note: Try and use `preloadAssetsTimeout` to exit for recursion, as against using a random number `maxImportUrls`